src/Controllers/UserController.php

```diff
@@ -35,31 +35,46 @@ public function setService(AbstractService $userService)
      */
     public function listAll(RequestInterface $request)
     {
+        // Check if user has permission
```

Comment:
😁 ok
the if is kinda self-explanatory.

However the controller methods are getting quite long, let's see if we can extract some functionality into separate classes